### PR TITLE
Support for PPK authentication via SFTP

### PR DESCRIPTION
**Pull request by ronaldheft-usds**
This PR adds PPK authentication to the SftpTransport.

Since Key Vault supports secrets as large as 25,000 bytes, PPK files are within that limit and can be stored as Azure secrets. In my testing, a PPK file stored in our JSON container averaged 2,800 bytes.

## Changes
- Refactors the SftpTransport to handle multiple authentication options
- Adds the ability to create UserPpk types via the CLI

## Checklist
- [x] Tested locally?
- [ ] Ran `quickTest all`?
- [ ] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from http://localhost:7071/api/download
- [ ] Updated the release notes? 
- [ ] Added tests?
- [ ] Did you check for sensitive data, and remove any? 
- [ ] Does logging contain sensitive data?  
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

## Security
*List potential security threats and mitigations in the PR*

## Fixes
*List GitHub issues this PR fixes*
- #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

### CODE REVIEW

File: prime-router/src/main/kotlin/transport/SftpTransport.kt [hunk 1]
```diff
@@ -67,33 +71,34 @@ class SftpTransport : ITransport {
...
         ): SSHClient {
             // create our client
             val sshClient = SSHClient()
             try {
                 sshClient.addHostKeyVerifier(PromiscuousVerifier())
                 sshClient.connect(host, port.toInt())
-                sshClient.authPassword(user, pass)
+                when (credential) {
+                    is UserPassCredential -> sshClient.authPassword(credential.user, credential.pass)
+                    is UserPpkCredential -> PuTTYKeyFile().init(StringReader(credential.key))
```
Comment (MauriceReeves-usds) [inline prime-router/src/main/kotlin/transport/SftpTransport.kt:99]: In this case, how does the ppk file get associated with the sshClient?

**Review by MauriceReeves-usds — CHANGES_REQUESTED** [review]
I pulled down the branch and pointed `gu-doh.elr` locally to Inductive Health's sftp server. I then saved the PPK as a credential and tried to execute the health check against their server but got an error: `Not Authenticated`. I'm not sure if that's because of the missing PPK password or it's not passing the PPK in.

I know the documentation for `sshj` is sparse, so if this is a pain to sort out, we can use `jsch` library instead which has more documentation and examples of how to use a PPK file when connecting over SFTP and SSH.

File: prime-router/src/main/kotlin/transport/SftpTransport.kt (hunk 1 above)
Comment (ronaldheft-usds) [reply prime-router/src/main/kotlin/transport/SftpTransport.kt]: Fixed

File: prime-router/src/main/kotlin/cli/CredentialsCli.kt [hunk 2]
```diff
@@ -66,7 +71,14 @@ class CredentialsCli : CredentialManagement, CliktCommand(
...
+sealed class UserTypeCredentialOptions(name: String) : CredentialConfig(name)
+
+class UserPassCredentialOptions : CredentialConfig("Options for credential type 'UserPass'") {
     val user by option().prompt(default = "")
     val pass by option().prompt(default = "", requireConfirmation = true)
+}
+
+class UserPpkCredentialOptions : CredentialConfig("Options for credential type 'UserPpk'") {
+    val user by option("--ppk-user").prompt(default = "")
+    val file by option("--ppk-file").file(mustExist = true).required()
```
[the remark this answers is not in the document]
Comment (ronaldheft-usds) [reply prime-router/src/main/kotlin/cli/CredentialsCli.kt]: Added support for an optional PPK password

**Review by MauriceReeves-usds — APPROVED** [review]
Passes testing locally. Well done.

**Pull request merged**